rules/jsx-a11y.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+module.exports = {
+    'rules': {
+        // Enforce emojis are wrapped in and provide screen reader access.
+        // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/accessible-emoji.md
+        'jsx-a11y/accessible-emoji': 2,
+        // Enforce all elements that require alternative text have meaningful information to relay back to end user.
+        // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/alt-text.md
+        'jsx-a11y/alt-text': 2,
+        // Enforce all anchors to contain accessible content.
+        // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-has-content.md
+        'jsx-a11y/anchor-has-content': 2,
+        // Enforce all anchors are valid, navigable elements.
+        // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.md
+        'jsx-a11y/anchor-is-valid': 2,
+        // Enforce elements with aria-activedescendant are tabbable.
+        // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/aria-activedescendant-has-tabindex.md
+        'jsx-a11y/aria-activedescendant-has-tabindex': 2,
+        // Enforce all aria-* props are valid.
+        // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/aria-props.m
+        'jsx-a11y/aria-props': 2,
+        // Enforce ARIA state and property values are valid.
+        // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/aria-proptypes.md
+        'jsx-a11y/aria-proptypes': 2,
+        // Enforce that elements with ARIA roles must use a valid, non-abstract ARIA role.
+        // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/aria-role.md
+        'jsx-a11y/aria-role': 2,
+        // Enforce that elements that do not support ARIA roles, states, and properties do not have those attributes.
+        // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/aria-unsupported-elements.md
+        'jsx-a11y/aria-unsupported-elements': 2,
+        // Enforce a clickable non-interactive element has at least one keyboard event listener.
+        // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/click-events-have-key-events.md
+        'jsx-a11y/click-events-have-key-events': 0,
```

Comment:
Does this rule cover scenarios such the one below?

```html
<ol>
   <li role="button" onClick={...} />
   ...
<ol>
```

If it does, then we should probably activate it, no?

Reply:
This rule would cover that scenario, yes. However the current implementation of these rules would prevent even having a `button` role in an `<li>` element, because of the [`no-noninteractive-element-to-interactive-role`](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/no-noninteractive-element-to-interactive-role.md). 

Original conversation on this rule deemed that it should at most be a warning. 

Personally, i think the rule should be turned on only after also considering and making an effort at standardizing the expected key events.

Reply:
Could you please explain why are we enabling [`no-noninteractive-element-to-interactive-role`](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/no-noninteractive-element-to-interactive-role.md) as an error? There are valid use cases to use, e.g., interactive `li` tags. Following what you said above, I agree we should make this rule a warning at most.

Reply:
In my opinion, [click-events-have-events](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/click-events-have-key-events.md) must be enabled as an error. Having a key event handler is a basic accessibility requirement for keyboard users.

Reply:
What are we going to do about this? 🤔

Reply:
@acostalima I might be wrong, but whenever you click enter or space in a clickable element, it will trigger a click event.

Reply:
If I stand correct, enabling this rule is too harsh for most situations. We just need to ensure that all elements with click are focusable, so that users can use tab to select them.

Reply:
@satazor Following the example in my comment above, I'm not so sure about elements where the `role="button"` is specified. Source: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#Required_event_handlers

Reply:
In those elements, the `tabIndex` should be specified, so that the elements are targetable via keyboard. @threequartersjohn do we enforce this via other rules?

Reply:
No, actually we enforce that no `tabIndex` is specified in non interactive components ([`no-noninteractive-tabindex`](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/no-noninteractive-tabindex.md)), as we also enforce that no non-interactive component is given an interaction or interactive role ([`no-noninteractive-element-interactions`](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/no-noninteractive-element-interactions.md), [`no-noninteractive-element-to-interactive-role`](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/no-noninteractive-element-to-interactive-role.md)). 

In practice, there would be no situation where you'd need a `tabIndex` in an element like a `<div>` because you wouldn't make the `<div>` an interactive element. Please refer to the provided links for the reasoning behind each rule.

Reply:
@threequartersjohn I think you misunderstood my comment. I was referring to interactive elements and not non-interactive ones.

A div with a click event is an interactive element that should be focusable. That’s where tabindex come in: by defining a tabindex, it becomes reachable via keyboard. Because a div with a tab index can be reached via keyboard, pressing enter on it will fire a click event (please see link mentioned by @acostalima).

Having that said, we should enforce that elements with onClick handlers have tabindex, except if they are buttons, anchors or other elements that are already focusable. So, if we use tabindex, having keyboard handlers is effectively unnecessary / redundant.

Reply:
After some tests, i'll turn on [click-events-have-events](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/click-events-have-key-events.md).

This is because adding a `tabIndex`, a `button` role _and_ an `onclick` event, it still will not fire the `onclick` event through keyboard interaction as it does in buttons and similar elements.

Reply:
@threequartersjohn 👌

Reply:
Resolving as this rule was activated as a warning.